Examples/Fermions/fermi_hubbard.py

```diff
@@ -0,0 +1,91 @@
+import netket as nk
+import numpy as np
+import matplotlib.pyplot as plt
+import json
+import jax
+
+from netket import experimental as nkx
+
+L = 2  # take a 2x2 lattice
+D = 2
+t = 1  # tunneling/hopping
+U = 0.01  # coulomb
+
+# create the graph our fermions can hop on
+g = nk.graph.Hypercube(length=L, n_dim=D, pbc=True)
+Nsites = g.n_nodes
+
+# create a hilbert space with 2 up and 2 down spins
+hi = nkx.hilbert.SpinOrbitalFermions(Nsites, s=1 / 2, n_fermions=(2, 2))
+
+# create an operator representing fermi hubbard interactions
+# -t (i^ j + h.c.) + U (i^ i j^ j)
+# we will create a helper function to abbreviate the creation, destruction and number operators
+# each operator has a site and spin projection (sz) in order to find the right position in the hilbert space samples
+def c(site, sz):
+    return nkx.operator.fermion.create(hi, site, sz=sz)
+
+
+def cdag(site, sz):
+    return nkx.operator.fermion.destroy(hi, site, sz=sz)
+
+
+def nc(site, sz):
+    return nkx.operator.fermion.number(hi, site, sz=sz)
+
+
+up = +1 / 2
+down = -1 / 2
+ham = []
+for sz in (up, down):
+    for u, v in g.edges():
+        ham.append(-t * cdag(u, sz) * c(v, sz) - t * cdag(v, sz) * c(u, sz))
+for u in g.nodes():
+    ham.append(U * nc(u, up) * nc(u, down))  # spinful interaction on site
+ham = sum(ham)
```

Comment:
can't you ham=0, ham+= ?

Reply:
Now I can (see most recent version and rewrite)